### ISSUE TITLE
player texture brightens on damage

### ISSUE THREAD

**Issue by DS-Minetest** [minetest/minetest#5739]
It's a bug that many people experienced already. The player's texture becomes brighter and brighter. I thought, a mod would cause it but no. Probably this also happens with any other entity. Normally the brightening effect should wear off after ~1 second. But if you get much times damage at once (eg. from many mobs or radiation), it doesn't.
<details>
<summary>some links, that were given to me</summary>

https://github.com/minetest/minetest/blob/master/src/content_cao.cpp#L1759
https://github.com/minetest/minetest/blob/master/src/content_cao.cpp#L1311
https://github.com/minetest/minetest/blob/master/src/content_cao.cpp#L1233
https://github.com/minetest/minetest/commit/c57b4f
</details>

**Comment by Fixer-007** [comment]
I think I've seen that too

**Comment by ExeterDad** [comment]
I can confirm this as well. I thought it was related to our skins mod so I didn't think much of it. I've also noticed after a player is killed, They appear white.

**Comment by paramat** [comment]
Didi it perhaps go red a while ago and now goes pale? I've noticed this too it seemed weird.

**Comment by GreenXenith** [comment]
It has never gone red in the past. It has always gone white with a smoke puff.

**Comment by Calinou** [comment]
This is not really a bug (at least, I think), though the implementation can lead to confusion – since March 2012 or so, players being hit will flash white, the duration will depend based on the damage dealt. Since somewhere in 2015 or 2016, players receiving any damage will flash white, not just players you hit. (Also, they will flash white if you hit them even if PvP is disabled.)

A smoke puff also appears when you kill any entity directly, including a player. It's just a single, non-animated texture that stays present for 1 second (it can also appear several times if you "kill" the entity more than once due to lag).

**Comment by DS-Minetest** [comment]
@Calinou The bug is that the texture doesn't become normal again sometimes.

**Comment by kilbith** [comment]
I can reproduce it. After a serie of hits : 

![screenshot_20170517_152258](https://cloud.githubusercontent.com/assets/7883281/26156157/beed0d62-3b15-11e7-828a-0a4fcc3fde0a.png)

**Comment by stujones11** [comment]
Fix #5767

**Comment by lhofhansl** [comment]
Looks like this is happening again now.
Not exactly sure when it started. Made a feeble attempt to track it down, but didn't find the culprit, yet.

**Comment by lhofhansl** [comment]
OK... Figured it out. The modifier is passed as const std::string &mod. It should be passed by value; otherwise we may pass the same reference around. The comment *right* above the definition in content_cao.h even states that.

**Comment by lhofhansl** [comment]
Will file a new PR, or we can make the two line change in place. Fine either way. :)

**Comment by DS-Minetest** [comment]
#7332 was merged, this can be closed again.

**Comment by SmallJoker** [comment]
Re-opening after #8554.
https://github.com/minetest/minetest/pull/7332/files#diff-b396c7bb7b51d0369c47a0f7df57434bR201 :thinking:

**Comment by LoneWolfHT** [comment]
Could this be fixed? Or is the fix a little more complex than I think it'll be?

**Comment by SmallJoker** [comment]
a1459a9

**Comment by ClobberXD** [comment]
I don't think this issue has been fixed yet. I faced this issue quite a few times today. MT is at 91d244c (latest at the time of compiling today). Strangely, I haven't encountered this bug ever before. Ironically, the bug started showing up only after I updated to a1459a9. Strangely, this happens only on a local server, where I run two instances of MT, one being the server+client and the other being the client.

#### Steps to reproduce

- Kill a player on a local server.
- Observe that their skin isn't restored to normal upon the last hit, when they die.
- Observe that their skin is permanently white from then (the first death) onwards.

I'll try to make a video soon. Please re-open this issue.

**Comment by LoneWolfHT** [comment]
This was fixed for me

**Comment by SmallJoker** [comment]
Ensure it's a clean build, so that any build issues can be excluded as possible cause. The patch worked fine, fixed the issue for me.

**Comment by ClobberXD** [comment]
Can't reproduce with a clean build after numerous attempts. Sorry for the trouble. This issue can be closed.
